src/editpage/components/assistance/SettingsModal.vue

```diff
@@ -0,0 +1,232 @@
+            <input v-model="zoomSliderValue" type="range" min="0.3" max="3"
+            step="0.1"
+            value="50" class="slider" id="myRange" @change="zoomSliderChanged">
+        </div> -->
+        <table class='padding-table-columns'>
+          <tr>
+            <th><h5 style='padding-right: 50px;'>Change the zoom level</h5></th>
+            <th>
+              <button class='settings-modal-button' @click="zoomIn">
+                Zoom in
+              </button>
+              &nbsp;
+              <button class='settings-modal-button' @click="zoomOut">
+                Zoom out
+              </button>
+            </th>
+          </tr>
+          <tr>
+            <th>
+              <h5 style='padding-right: 50px;'>Theme</h5>
+            </th>
+            <th>
+              <div class="dropdown">
+                <button class="dropbtn settings-modal-button">Select</button>
+                <div class="dropdown-content">
+                  <a v-for="style in styles" :key="style"
+                      @click="changeTheme(style)">
+                      {{style}}</a>
+                </div>
+              </div>
+            </th>
+          </tr>
+        </table>
+
+      </div>
+    </div>
+</template>
+
+
+<script>
+import Vue from 'vue';
+
+export default {
+  name: 'SettingsModal',
+  components: {},
+  props: {},
+  data: function() {
+    return {
+      eventBus: new Vue(),
+      zoomMin: 0.3,
+      zoomMax: 3,
+      zoomSliderValue: 0,
+      styles: ['light', 'dark', 'light_roboto', 'dark_roboto'],
+    };
+  },
+  methods: {
+    forwardEvent: function(event) {
+      console.log(event);
+      this.eventBus.$emit(event);
+      console.log('forwarded done');
+    },
+    closeSettingsModal: function() {
+      document.getElementById('settingsModal').style.display = 'none';
+    },
+
+    openSettingsModal: function() {
+      document.getElementById('settingsModal').style.display = 'block';
+      this.updateConfigurationString();
+      // const release = await store.state.lock.acquire();
+
+      // release();
+    },
+
+    zoomIn: function() {
+      this.zoomChange(1.1);
+    },
+
+    zoomOut: function() {
+      this.zoomChange(1/1.1);
+    },
+
+    zoomSliderChanged: function() {
+      const val = parseFloat(this.zoomSliderValue);
+      this.$store.commit('setZoom', val);
+    },
+
+    zoomChange: function(factor) {
+      const newZoom = this.$store.state.settings.zoom * factor;
+
+      const boundedZoom = Math.max(0.3, Math.min(newZoom, 3.0));
+      this.$store.commit('setZoom', boundedZoom);
+      this.updateConfigurationString();
+    },
+
+    changeTheme: function(theme) {
+      if (this.styles.includes(theme)) {
+        this.$store.commit('setTheme', theme);
+      }
+    },
+
+    updateConfigurationString: function() {
+      const libs = this.$store.state.libraries;
+      const settings = this.$store.state.settings;
+      document.getElementById('settingsOverview').innerHTML =
```

Comment:
Why are we creating this html manually?

Reply:
I agree it's quite ugly. Any suggestions? I want to use the v-for but then libs has to be in the data somehow...